# PATTERN PROGRAM 

In [1]:
#### USING FOR LOOP AND IF-ELSE CONDITION .AND ALSO WITH THE HELP OF BOOLEAN OPERATOR WE CHECK THE CONDITION IF ALL SATIFIED THEN PRINT REQUIRED PATTERN
for i in range(5):
    for j in range(5):
        if i==0 or i==4 or j==0 or j==4:
             print("*",end=" ")
        else:
            print(end=" ")
    print()        
          

* * * * * 
*    * 
*    * 
*    * 
* * * * * 


In [2]:
for i in range(5):
    for j in range(5):
        if i==4 or i-j==0 or j<=0:
            print("*",end="")
        else:
            print(end=" ")
    print()        

*    
**   
* *  
*  * 
*****


# TIME MODULE-IT HANDEL TIME RELATED TASK .WE KNOW THAT EVERY PROGRAM TAKE EXECUTION TIME BUT HOW MUCH THAT WE DONT KNOW.TO RESOLVE THIS WE USE TIME MODULE.

In [3]:
### IMPORT LIBRARIES
import time
## here we create instances of time.time()
initial=time.time()
### print hey jack through while loop
k=0
while(k<15):
    print("hey jack")
    k+=1
#### here we check execution time of program that how much time taken by while loop     
print("While loop run in",time.time(),"seconds") 



hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
hey jack
While loop run in 1602657872.9360678 seconds


# THE FUNCTION time.time() returns the current system time in ticks .

In [4]:
#### IT GIVE LOCAL CURRENT TIME
import time
localtime=time.asctime()
localtime

'Wed Oct 14 12:14:33 2020'

In [5]:
###IMPORT LIBRARY
import time
### CREATE OBJECT OF TIME.TIME()
initial1=time.time()
### WHILE LOOP
k=0
while(k<15):
    print("python")
#### IT IS A TIME.SLEEP FUNCTION AND WE HAVE GIVEN SPECIFIC NUMBER     
    time.sleep(2) 
    k+=1
###HERE WE HAVE CHECK THE EXECUTION TIME OF WHILE LOOP AND IN MEAN TIME ALSO IT TAKING THE DELAY FOR SPECIFIC SECONDS
print("while loop run in",time.time()-initial1,"seconds")

### CREATE ANOTHER INSTANCES OF TIME.TIME()
initial2=time.time()
#### FOR LOOP
for i in range(15):
    print("harry")
#### HERE WE HAVE CHECK EXECUTION TIME OF FOR LOOP WITHOUT ANY DELAY OF COMMANDS    
print("for loop run in",time.time()-initial2,"seconds")    
    
    

python
python
python
python
python
python
python
python
python
python
python
python
python
python
python
while loop run in 30.186053037643433 seconds
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
harry
for loop run in 0.0 seconds


# WHILE LOOP TAKEN EXECUTION TIME OF 30.186053037643433 SECONDS BECAUSE IT  TAKE DELAY OF 2 SEC AFTER  EVERY COMMANDS.BUT FOR LOOP EXECUTION TIME IS 0.0 SECONDS WITHOUT ANY DELAY OF SECONDS 